earthly

```diff
@@ -38,10 +38,13 @@ detect_frontend() {
     return
   fi
 
-  echo "none"
+  echo >&2 "failed to detect docker/podman frontend"
+  exit 1
 }
 
-frontend_bin=$(detect_frontend)
+if [ "$EARTHLY_DISABLE_AUTO_UPDATE" != "true" ]; then
+    frontend_bin=$(detect_frontend)
+fi
```

Comment:
Is this intended? We should still detect the frontend even if auto-update is disabled.

Reply:
Oh nvm - I see the logic now :-)

Reply:
it's needed for our `+earthly-script-no-stdout` test.

Reply:
ah but there's a case where calling `do_reset()` will skip this part if a user has `EARTHLY_DISABLE_AUTO_UPDATE=true`.

I'll introduce a new `EARTHLY_DISABLE_FRONTEND_DETECTION` and add a comment saying it's meant for tests.